src/jit/importer.cpp

```diff
@@ -792,7 +787,7 @@ void                Compiler::impAssignTempGen(unsigned     tmpNum,
 {
     GenTreePtr asg;
 
-    if (varTypeIsStruct(val))
+    if ((val->TypeGet() == TYP_STRUCT) || varTypeIsSIMD(val))
```

Comment:
I don't know why I made this change. I have undone it. It is semantically the same, at least for now, and even in future when we may have other struct types, it would always be safest to call lvaSetStruct on them.